Update the developer's guide with new copyright hook (#5848)
rapidsai/cuml: wiki/cpp/DEVELOPER_GUIDE.md

```

#### Copyright header

> RAPIDS' [pre-commit-hooks](https://github.com/rapidsai/pre-commit-hooks) checks the Copyright

Reviewer: ```suggestion
RAPIDS [pre-commit-hooks](https://github.com/rapidsai/pre-commit-hooks) check the copyright
```

header for all git-modified files.

Manually, you can run the following to bulk-fix the header on all files in the repository:
```bash
pre-commit run -a verify-copyright
```

> Keep in mind that this only applies to files tracked by git and having been modified.

Reviewer: ```suggestion
Keep in mind that this only applies to files tracked by git that have been modified.
```
